lib/spack/spack/cmd/arch.py

```diff
@@ -35,5 +41,11 @@ def arch(parser, args):
         print(arch.os)
     elif args.target:
         print(arch.target)
+    elif args.frontend:
```

Comment:
@tkameyama 

If the user specifies `--target` and `--backend`, I think we also want to support that. So I think a good approach would be to set `arch` based on whether a user specifies `--frontend` or `--backend` (or neither), and leave the rest of the function as is:

```
def arch(...
    if args.frontend:
        arch = architecture.Arch(architecture.platform(), 'fe', 'fe'))
    elif args.backend:
        ...
    else:
        arch = architecture.Arch(architecture.platform(), 'default_os', 'default_target')
    ...
```

Reply:
I rewrite so,